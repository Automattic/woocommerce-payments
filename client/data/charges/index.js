/** @format */

/**
 * Internal dependencies
 */
import reducer from './reducer.gen';
import * as selectors from './selectors.gen';
import * as actions from './actions.gen';
import * as resolvers from './resolvers';

export { reducer, selectors, actions, resolvers };
export * from './hooks.gen';
