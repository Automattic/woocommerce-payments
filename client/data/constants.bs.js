// Generated by BUCKLESCRIPT, PLEASE EDIT WITH CARE


var namespace = "/wc/v3/payments";

var storeName = "wc/payments";

export {
  namespace ,
  storeName ,
  
}
/* No side effect */
