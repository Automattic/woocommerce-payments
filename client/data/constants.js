/** @format */

export const NAMESPACE = '/wc/v3/payments';
export const STORE_NAME = 'wc/payments';

export const ID_PREFIX = {
	transactions: 'transactions',
};
