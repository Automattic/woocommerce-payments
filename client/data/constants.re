let namespace = "/wc/v3/payments";
let storeName = "wc/payments";
