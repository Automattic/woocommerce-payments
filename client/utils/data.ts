/** @format */

export const getResourceId = (
	identifier: Record< string, string >
): string => {
	return JSON.stringify( identifier, Object.keys( identifier ).sort() );
};
