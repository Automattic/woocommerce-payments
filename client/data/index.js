/** @format */

/**
 * Internal dependencies
 */
import { STORE_NAME } from './constants';
import { initStore } from './store';

initStore();

export const WCPAY_STORE_NAME = STORE_NAME;

// We only ask for hooks when importing directly from 'data'.
export * from './deposits/hooks';
export * from './transactions/hooks';
export * from './charges/hooks';
export * from './timeline/hooks';
export * from './disputes/hooks';
export * from './settings/hooks';
export * from './multi-currency';
