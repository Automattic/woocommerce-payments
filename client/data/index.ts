/** @format */

/**
 * Internal dependencies
 */
import { STORE_NAME } from './constants';
import { initStore } from './store';

initStore();

// eslint-disable-next-line @typescript-eslint/naming-convention
export const WCPAY_STORE_NAME = STORE_NAME;

// We only ask for hooks when importing directly from 'wcpay/data'.
export * from './deposits/hooks';
export * from './transactions/hooks';
export * from './charges/hooks';
export * from './timeline/hooks';
export * from './disputes/hooks';
export * from './settings/hooks';
export * from './multi-currency';
export * from './card-readers/hooks';
export * from './capital/hooks';
export * from './documents/hooks';
export * from './payment-intents/hooks';
export * from './authorizations/hooks';
