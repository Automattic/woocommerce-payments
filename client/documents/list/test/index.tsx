/** @format */

/**
 * External dependencies
 */
import * as React from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';
import { getQuery } from '@woocommerce/navigation';

/**
 * Internal dependencies
 */
import { DocumentsList } from '../';
import { useDocuments, useDocumentsSummary } from 'data/index';
import type { Document } from 'data/documents/hooks';

jest.mock( 'data/index', () => ( {
	useDocuments: jest.fn(),
	useDocumentsSummary: jest.fn(),
} ) );

const mockUseDocuments = useDocuments as jest.MockedFunction<
	typeof useDocuments
>;

const mockUseDocumentsSummary = useDocumentsSummary as jest.MockedFunction<
	typeof useDocumentsSummary
>;

const getMockDocuments: () => Document[] = () => [
	{
		document_id: 'test_document_123456',
		date: '2020-01-02 17:46:02',
		type: 'test_document',
		period_from: '',
		period_to: '',
	},
	{
		document_id: 'test_document_654321',
		date: '2020-01-05 04:22:59',
		type: 'test_document',
		period_from: '',
		period_to: '',
	},
];

describe( 'Documents list', () => {
	let container: Element;
	let rerender: ( ui: React.ReactElement ) => void;
	beforeEach( () => {
		mockUseDocuments.mockReturnValue( {
			documents: getMockDocuments(),
			isLoading: false,
			documentsError: undefined,
		} );

		mockUseDocumentsSummary.mockReturnValue( {
			documentsSummary: {
				count: 10,
			},
			isLoading: false,
		} );

		( { container, rerender } = render( <DocumentsList /> ) );
	} );

	function expectSortingToBe( field: string, direction: string ) {
		expect( getQuery().orderby ).toEqual( field );
		expect( getQuery().order ).toEqual( direction );
		const useDocumentsCall =
			mockUseDocuments.mock.calls[
				mockUseDocuments.mock.calls.length - 1
			];
		expect( useDocumentsCall[ 0 ].orderby ).toEqual( field );
		expect( useDocumentsCall[ 0 ].order ).toEqual( direction );
	}

	function sortBy( field: string ) {
		user.click( screen.getByRole( 'button', { name: field } ) );
		rerender( <DocumentsList /> );
	}

	test( 'renders correctly', () => {
		expect( container ).toMatchSnapshot();
	} );

	test( 'sorts by default field date', () => {
		sortBy( 'Date and time' );
		expectSortingToBe( 'date', 'asc' );

		sortBy( 'Date and time' );
		expectSortingToBe( 'date', 'desc' );
	} );

	test( 'renders table summary only when the documents summary data is available', () => {
		mockUseDocumentsSummary.mockReturnValue( {
			documentsSummary: {},
			isLoading: true,
		} );

		( { container } = render( <DocumentsList /> ) );
		let tableSummary = container.querySelectorAll(
			'.woocommerce-table__summary'
		);
		expect( tableSummary ).toHaveLength( 0 );

		mockUseDocumentsSummary.mockReturnValue( {
			documentsSummary: {
				count: 10,
			},
			isLoading: false,
		} );

		( { container } = render( <DocumentsList /> ) );
		tableSummary = container.querySelectorAll(
			'.woocommerce-table__summary'
		);

		expect( tableSummary ).toHaveLength( 1 );
	} );

	test( 'renders table summary only when the documents summary data is available with a single document', () => {
		mockUseDocumentsSummary.mockReturnValue( {
			documentsSummary: {},
			isLoading: true,
		} );

		( { container } = render( <DocumentsList /> ) );
		let tableSummary = container.querySelectorAll(
			'.woocommerce-table__summary'
		);
		expect( tableSummary ).toHaveLength( 0 );

		mockUseDocumentsSummary.mockReturnValue( {
			documentsSummary: {
				count: 1,
			},
			isLoading: false,
		} );

		( { container } = render( <DocumentsList /> ) );
		tableSummary = container.querySelectorAll(
			'.woocommerce-table__summary'
		);

		expect( tableSummary ).toHaveLength( 1 );
		expect( container ).toMatchSnapshot();
	} );
} );
