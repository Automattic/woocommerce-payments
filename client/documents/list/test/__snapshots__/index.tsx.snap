// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents list renders correctly 1`] = `
<div>
  <div
    class=" woocommerce-payments-page"
  >
    <div
      class="woocommerce-filters-documents"
    >
      <h2
        class="screen-reader-text"
      >
        Filters
      </h2>
      <div
        class="woocommerce-filters"
      >
        <div
          class="woocommerce-filters__basic-filters"
        >
          <div
            class="woocommerce-filters-filter"
          >
            <span
              class="woocommerce-filters-label"
            >
              Show
              :
            </span>
            <div
              class="components-dropdown"
            >
              <button
                aria-expanded="false"
                class="components-button woocommerce-dropdown-button"
                type="button"
              >
                <div
                  class="woocommerce-dropdown-button__labels"
                >
                  <span>
                    All documents
                  </span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="woocommerce-card woocommerce-table woocommerce-analytics__card documents-list woocommerce-report-table has-search has-menu has-action"
    >
      <div
        class="woocommerce-card__header"
      >
        <div
          class="woocommerce-card__title-wrapper"
        >
          <h2
            class="woocommerce-card__title woocommerce-card__header-item"
          >
            Documents
          </h2>
        </div>
        <div
          class="woocommerce-card__action woocommerce-card__header-item"
        />
        <div
          class="woocommerce-card__menu woocommerce-card__header-item"
        >
          <div
            class="woocommerce-ellipsis-menu"
          >
            <div
              class="components-dropdown"
            >
              <button
                aria-expanded="false"
                class="components-button woocommerce-ellipsis-menu__toggle has-text has-icon"
                title="Choose which values to display"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-ellipsis"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 10c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm12-2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-7 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 9v1.5h16V9H4zm12 5.5h4V13h-4v1.5zm-6 0h4V13h-4v1.5zm-6 0h4V13H4v1.5z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="woocommerce-card__body"
      >
        <div
          aria-hidden="false"
          aria-labelledby="caption-0"
          class="woocommerce-table__table"
          role="group"
        >
          <table>
            <caption
              class="woocommerce-table__caption screen-reader-text"
              id="caption-0"
            >
              Documents
            </caption>
            <tbody>
              <tr>
                <th
                  aria-sort="descending"
                  class="woocommerce-table__header date is-left-aligned is-sortable is-sorted"
                  role="columnheader"
                  scope="col"
                >
                  <button
                    aria-describedby="header-0-0"
                    class="components-button"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
                      />
                    </svg>
                    <span
                      aria-hidden="true"
                    >
                      Date
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      Date and time
                    </span>
                  </button>
                  <span
                    class="screen-reader-text"
                    id="header-0-0"
                  >
                    Sort by Date and time in ascending order
                  </span>
                </th>
                <th
                  class="woocommerce-table__header is-left-aligned"
                  role="columnheader"
                  scope="col"
                >
                  <span
                    aria-hidden="true"
                  >
                    Type
                  </span>
                  <span
                    class="screen-reader-text"
                  >
                    Type
                  </span>
                </th>
                <th
                  class="woocommerce-table__header"
                  role="columnheader"
                  scope="col"
                >
                  <span
                    aria-hidden="true"
                  >
                    Description
                  </span>
                  <span
                    class="screen-reader-text"
                  >
                    Description
                  </span>
                </th>
                <th
                  class="woocommerce-table__header is-numeric"
                  role="columnheader"
                  scope="col"
                >
                  <span
                    aria-hidden="true"
                  />
                  <span
                    class="screen-reader-text"
                  >
                    Download
                  </span>
                </th>
              </tr>
              <tr>
                <th
                  class="woocommerce-table__item date is-left-aligned is-sorted"
                  scope="row"
                >
                  Jan 2, 2020
                </th>
                <td
                  class="woocommerce-table__item is-left-aligned"
                >
                  VAT Invoice
                </td>
                <td
                  class="woocommerce-table__item"
                >
                  VAT invoice without proper period dates
                </td>
                <td
                  class="woocommerce-table__item is-numeric"
                >
                  <a
                    href="https://site.com/wp-json/wc/v3/payments/documents/vat_invoice_123456?_wpnonce=random_wp_rest_nonce"
                    rel="noopener noreferrer"
                    style="display: inline;"
                    target="_blank"
                  >
                    Download
                  </a>
                </td>
              </tr>
              <tr>
                <th
                  class="woocommerce-table__item date is-left-aligned is-sorted"
                  scope="row"
                >
                  Jan 4, 2020
                </th>
                <td
                  class="woocommerce-table__item is-left-aligned"
                >
                  VAT Invoice
                </td>
                <td
                  class="woocommerce-table__item"
                >
                  VAT invoice without proper period dates
                </td>
                <td
                  class="woocommerce-table__item is-numeric"
                >
                  <a
                    href="https://site.com/wp-json/wc/v3/payments/documents/vat_invoice_654321?_wpnonce=random_wp_rest_nonce"
                    rel="noopener noreferrer"
                    style="display: inline;"
                    target="_blank"
                  >
                    Download
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <ul
          class="woocommerce-table__summary"
        >
          <li
            class="woocommerce-table__summary-item"
          >
            <span
              class="woocommerce-table__summary-value"
            >
              10
            </span>
            <span
              class="woocommerce-table__summary-label"
            >
              documents
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Documents list renders table summary only when the documents summary data is available with a single document 1`] = `
<div>
  <div
    class=" woocommerce-payments-page"
  >
    <div
      class="woocommerce-filters-documents"
    >
      <h2
        class="screen-reader-text"
      >
        Filters
      </h2>
      <div
        class="woocommerce-filters"
      >
        <div
          class="woocommerce-filters__basic-filters"
        >
          <div
            class="woocommerce-filters-filter"
          >
            <span
              class="woocommerce-filters-label"
            >
              Show
              :
            </span>
            <div
              class="components-dropdown"
            >
              <button
                aria-expanded="false"
                class="components-button woocommerce-dropdown-button"
                type="button"
              >
                <div
                  class="woocommerce-dropdown-button__labels"
                >
                  <span>
                    All documents
                  </span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="woocommerce-card woocommerce-table woocommerce-analytics__card documents-list woocommerce-report-table has-search has-menu has-action"
    >
      <div
        class="woocommerce-card__header"
      >
        <div
          class="woocommerce-card__title-wrapper"
        >
          <h2
            class="woocommerce-card__title woocommerce-card__header-item"
          >
            Documents
          </h2>
        </div>
        <div
          class="woocommerce-card__action woocommerce-card__header-item"
        />
        <div
          class="woocommerce-card__menu woocommerce-card__header-item"
        >
          <div
            class="woocommerce-ellipsis-menu"
          >
            <div
              class="components-dropdown"
            >
              <button
                aria-expanded="false"
                class="components-button woocommerce-ellipsis-menu__toggle has-text has-icon"
                title="Choose which values to display"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-ellipsis"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 10c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm12-2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-7 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 9v1.5h16V9H4zm12 5.5h4V13h-4v1.5zm-6 0h4V13h-4v1.5zm-6 0h4V13H4v1.5z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="woocommerce-card__body"
      >
        <div
          aria-hidden="false"
          aria-labelledby="caption-7"
          class="woocommerce-table__table"
          role="group"
        >
          <table>
            <caption
              class="woocommerce-table__caption screen-reader-text"
              id="caption-7"
            >
              Documents
            </caption>
            <tbody>
              <tr>
                <th
                  aria-sort="descending"
                  class="woocommerce-table__header date is-left-aligned is-sortable is-sorted"
                  role="columnheader"
                  scope="col"
                >
                  <button
                    aria-describedby="header-7-0"
                    class="components-button"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
                      />
                    </svg>
                    <span
                      aria-hidden="true"
                    >
                      Date
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      Date and time
                    </span>
                  </button>
                  <span
                    class="screen-reader-text"
                    id="header-7-0"
                  >
                    Sort by Date and time in ascending order
                  </span>
                </th>
                <th
                  class="woocommerce-table__header is-left-aligned"
                  role="columnheader"
                  scope="col"
                >
                  <span
                    aria-hidden="true"
                  >
                    Type
                  </span>
                  <span
                    class="screen-reader-text"
                  >
                    Type
                  </span>
                </th>
                <th
                  class="woocommerce-table__header"
                  role="columnheader"
                  scope="col"
                >
                  <span
                    aria-hidden="true"
                  >
                    Description
                  </span>
                  <span
                    class="screen-reader-text"
                  >
                    Description
                  </span>
                </th>
                <th
                  class="woocommerce-table__header is-numeric"
                  role="columnheader"
                  scope="col"
                >
                  <span
                    aria-hidden="true"
                  />
                  <span
                    class="screen-reader-text"
                  >
                    Download
                  </span>
                </th>
              </tr>
              <tr>
                <th
                  class="woocommerce-table__item date is-left-aligned is-sorted"
                  scope="row"
                >
                  Jan 2, 2020
                </th>
                <td
                  class="woocommerce-table__item is-left-aligned"
                >
                  VAT Invoice
                </td>
                <td
                  class="woocommerce-table__item"
                >
                  VAT invoice without proper period dates
                </td>
                <td
                  class="woocommerce-table__item is-numeric"
                >
                  <a
                    href="https://site.com/wp-json/wc/v3/payments/documents/vat_invoice_123456?_wpnonce=random_wp_rest_nonce"
                    rel="noopener noreferrer"
                    style="display: inline;"
                    target="_blank"
                  >
                    Download
                  </a>
                </td>
              </tr>
              <tr>
                <th
                  class="woocommerce-table__item date is-left-aligned is-sorted"
                  scope="row"
                >
                  Jan 4, 2020
                </th>
                <td
                  class="woocommerce-table__item is-left-aligned"
                >
                  VAT Invoice
                </td>
                <td
                  class="woocommerce-table__item"
                >
                  VAT invoice without proper period dates
                </td>
                <td
                  class="woocommerce-table__item is-numeric"
                >
                  <a
                    href="https://site.com/wp-json/wc/v3/payments/documents/vat_invoice_654321?_wpnonce=random_wp_rest_nonce"
                    rel="noopener noreferrer"
                    style="display: inline;"
                    target="_blank"
                  >
                    Download
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <ul
          class="woocommerce-table__summary"
        >
          <li
            class="woocommerce-table__summary-item"
          >
            <span
              class="woocommerce-table__summary-value"
            >
              1
            </span>
            <span
              class="woocommerce-table__summary-label"
            >
              document
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
